lib/linux/sys/proctable/smaps.rb

```diff
@@ -0,0 +1,123 @@
+module Sys
+  class ProcTable
+    # Smaps represents a process' memory size for all mapped files
+    #
+    # A single mapped file memory entry looks like this:
+    #
+    #   00400000-004d4000 r-xp 00000000 fd:00 785                      /bin/bash
+    #   Size:                848 kB
+    #   Rss:                 572 kB
+    #   Pss:                 572 kB
+    #   Shared_Clean:          0 kB
+    #   Shared_Dirty:          0 kB
+    #   Private_Clean:       572 kB
+    #   Private_Dirty:         0 kB
+    #   Referenced:          572 kB
+    #   Anonymous:             0 kB
+    #   AnonHugePages:         0 kB
+    #   Swap:                  0 kB
+    #   KernelPageSize:        4 kB
+    #   MMUPageSize:           4 kB
+    #
+    # Have a look at `man 5 proc` on a linux distribution, to get some more
+    # information about the lines and fields in `/proc/[pid]/smaps`.
+    #
+    # Example:
+    #
+    #   smaps = Smaps.new(123, IO.read("/proc/1234/smaps")
+    #   smaps.pss  # => 109568
+    #   smaps.rss  # => 376832
+    #   smaps.uss  # => 98304
+    #   smaps.swap # => 196608
+    #   smaps.vss  # => 140034048
+    #
+    class Smaps
+
+      # Process ID for this smaps
+      attr_reader :pid
+
+      # Proportional set size
+      #
+      # PSS is the size of private pages added to each shared mapping's size
+      # divided by the number of processes that share it.  It is meant to
+      # provide a better representation of the amount of memory actually used
+      # by a process.
+      #
+      # If a process has 4k of private pages, 4k of shared pages shared with one
+      # other process, and 3k of pages shared with two other processes, the PSS
+      # is:
+      #
+      # 4k + (4k / 2) + (3k / 3) = 7k
+      #
+      attr_reader :pss
+      alias_method :proportional, :pss
```

Comment:
Hm, if we want the long hand, I think we should go all the way with proportional_set_size. Same for rss below.